python-backend/tests/permit/resources/test_permit_resource.py

```diff
@@ -1,29 +1,143 @@
 import json
-from tests.constants import TEST_PERMIT_GUID_1, TEST_MINE_GUID
+import uuid
+import pytest
+
+from app.api.mines.mine.models.mine import Mine
+from app.api.permits.permit.models.permit import Permit
+from app.api.permits.permit_amendment.models.permit_amendment import PermitAmendment
+from app.extensions import db
+from tests.constants import DUMMY_USER_KWARGS
+
+
+@pytest.fixture(scope="function")
+def setup_info(test_client):
+    test_permit_no = 'mx-test-231'
+    mine_1 = Mine.create_mine('1234567', 'TestMine', True, 'SW', DUMMY_USER_KWARGS)
+    mine_2 = Mine.create_mine('7654321', 'TestingMine', True, 'NW', DUMMY_USER_KWARGS)
+    mine_1.save()
+    mine_2.save()
+
+    permit = Permit.create(mine_2.mine_guid, test_permit_no, 'O', DUMMY_USER_KWARGS)
+    permit.save()
+
+    MINE_1_GUID = str(mine_1.mine_guid)
+    MINE_2_GUID = str(mine_2.mine_guid)
+    MINE_2_PERMIT_GUID = str(permit.permit_guid)
+    NON_EXISTENT_GUID = '8ef23184-02c4-4472-a912-380b5a0d9cae'
+
+    yield dict(
+        mine_1_guid=MINE_1_GUID,
+        mine_2_guid=MINE_2_GUID,
+        mine_2_permit_guid=MINE_2_PERMIT_GUID,
+        bad_guid=NON_EXISTENT_GUID,
+        in_use_permit_no=test_permit_no)
+
+    db.session.query(PermitAmendment).delete()
+    db.session.commit()
+    db.session.query(Permit).delete()
```

Comment:
Does this delete every entry in the table? 
Also, can this be done with one commit, or was this the FK issues you showed me?

Reply:
This does delete every entry in the table and it was the only way I found to get around the FK issue I was having.